### PR TITLE
Point of interest additional options

### PR DESCRIPTION
**Pull request by roosapi**
Add height, columns, and background image options for point of interest.

Swap title and id so that title becomes an argument and id an option (random id generated if one not given).

### CODE REVIEW

**Comment by Mankro** [comment]
The changes seem fine to me. I assume you have tested it thoroughly so that it works correctly.

General comments:

* The API of the directive changes here, so the material creators must be aware. This is even more important when this directive is used in two courses whose teachers will change now and they don't know these technical details very well.
* `STATIC_CONTENT_HOST` environment variable is required because we did not find a better way to fix the relative background image URL. Absolute URLs work in A+ and the environment variable enables building an absolute URL with the domain part.

**Comment by raphendyr** [comment]
As the API changes, there are two options:

a) add new directive, e.g. `.. point-of-interest2::`

b) The code needs to handle both interfaces (old and the new). So, the code could check that if there is a `:title:`, then the argument for the directive is the id. If there is no `:title:`, then the directive argument is the title and `:id:` is used or one is generated. If there is `:title:` and `:id:`, then an exception is raised.

**Comment by Mankro** [comment]
Indeed, handling backwards-compatibility would be important since people hate useless API changes, including myself. Raphendyr's idea sounds very reasonable and easy to implement as well.

**Comment by roosapi** [comment]
Fixed so that the old API is supported according to option B and squashed commits

File: directives/point_of_interest.py [hunk 1]
```diff
@@ -32,50 +40,106 @@ class PointOfInterest(Directive):
...
-        title_text = ''
-        if 'title' in  self.options:
+
+        if 'title' in self.options:
+            if 'id' in self.options:
+                raise SphinxError(u'Point of interest options can\'t contain both "title" and "id"; one of them should be provided as an argument instead')
+            name = self.arguments[0]
             title_text = self.options['title']
+        else:
+            name = self.options.get('id', ''.join(random.choice(string.ascii_lowercase + string.digits) for i in range(6)))
```
Comment (raphendyr) [inline directives/point_of_interest.py]: `.get('id') or random_string_generation` would be bit faster. Currently, the random string is always generated. Bit better looking code would be:

```
name = self.options.get('id')
if not name:
  choices = string.ascii_lowercase + string.digits
  name = ''.join(random.choice(choices) for i in range(6))
```

p.s. there is still change that the name is not unique.
p.s.s. `choices` is optimization, which is not done automatically by Python

File: directives/point_of_interest.py [hunk 2]
```diff
@@ -32,50 +40,106 @@ class PointOfInterest(Directive):
...
+            title_text = self.arguments[0]
```
Comment (raphendyr) [inline directives/point_of_interest.py]: Does this handle spaces in the string? e.g.

```
.. point-of-interest:: My long title with multiple words
```

File: directives/point_of_interest.py (hunk 1 above)
Comment (roosapi) [reply directives/point_of_interest.py]: I'll change that and add a check to make sure the id is unique. Not quite sure what you mean with the last one though?

File: directives/point_of_interest.py (hunk 2 above)
Comment (roosapi) [reply directives/point_of_interest.py]: Yes, spaces in the title are fine.

Comment (raphendyr) [reply directives/point_of_interest.py]: Good. I didn't remember how the arguments are split by sphinx.

File: directives/point_of_interest.py (hunk 1 above)
Comment (raphendyr) [reply directives/point_of_interest.py]: Just an information why I wrote the example like that. Real programming language would do the optimisation, but python can't.

Comment (roosapi) [reply directives/point_of_interest.py]: Ah ok, I see. Changed this and in the process noticed that poi info was not correctly cleared from the environment so added a fix for that as well.

File: directives/point_of_interest.py [hunk 3]
```diff
@@ -32,50 +40,114 @@ class PointOfInterest(Directive):
...
+
+        if 'title' in self.options:
+            if 'id' in self.options:
+                raise SphinxError(u'Point of interest options can\'t contain both "title" and "id"; one of them should be provided as an argument instead')
+            name = self.arguments[0]
             title_text = self.options['title']
+        else:
+            name = self.options.get('id')
+            if not name:
+              choices = string.ascii_lowercase + string.digits
```
Comment (Mankro) [inline directives/point_of_interest.py]: Indentation of only 2 spaces when it should be 4?

**Comment by raphendyr** [comment]
Looks good. I didn't review the html generation very well, but I presume old versions are not broken too much.

File: directives/point_of_interest.py (hunk 3 above)
Comment (roosapi) [reply directives/point_of_interest.py]: Hups, fixed!

**Comment by Mankro** [comment]
Are we ready to merge?

**Comment by raphendyr** [comment]
Fine by me.

**Pull request merged**